src/Elcodi/Bundle/ProductBundle/Tests/Functional/EventListener/RootCategoryEventListenerTest.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+/*
+ * This file is part of the Elcodi package.
+ *
+ * Copyright (c) 2014-2015 Elcodi.com
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * Feel free to edit as you please, and have fun.
+ *
+ * @author Marc Morera <yuhu@mmoreram.com>
+ * @author Aldo Chiecchia <zimage@tiscali.it>
+ * @author Elcodi Team <tech@elcodi.com>
+ */
+
+namespace Elcodi\Bundle\ProductBundle\Tests\Functional\Repository;
+
+use Doctrine\ORM\EntityManager;
+
+use Elcodi\Bundle\TestCommonBundle\Functional\WebTestCase;
+use Elcodi\Component\Product\Entity\Interfaces\CategoryInterface;
+use Elcodi\Component\Product\Factory\CategoryFactory;
+use Elcodi\Component\Product\Repository\CategoryRepository;
+
+/**
+ * Class CategoryRepositoryTest
+ */
+class RootCategoryEventListenerTest extends WebTestCase
+{
+    /**
+     * @var CategoryRepository
+     *
+     * LocationProvider class
+     */
+    protected $categoryRepository;
+
+    /**
+     * @var CategoryFactory
+     *
+     * Category factory class
+     */
+    protected $categoryFactory;
+
+    /**
+     * @var EntityManager
+     *
+     * Category entity manager
+     */
+    protected $entityManager;
+
+    /**
+     * Setup
+     */
+    public function setUp()
+    {
+        parent::setUp();
+
+        $this->categoryRepository = $this
```

Comment:
In one line. Splitting each `->` in several lines has sense when `count("->") >= 3`